hackathon_site/dashboard/frontend/src/components/teamDetail/TeamPendingOrderTable/TeamPendingOrderTable.tsx

```diff
@@ -0,0 +1,309 @@
+                                        elevation={2}
+                                        square={true}
+                                    >
+                                        <Table className={styles.table} size="small">
+                                            <TableHead>
+                                                <TableRow>
+                                                    <TableCell
+                                                        className={styles.widthFixed}
+                                                    />
+                                                    <TableCell
+                                                        className={styles.width6}
+                                                    >
+                                                        Name
+                                                    </TableCell>
+                                                    <TableCell
+                                                        className={`${styles.width1} ${styles.noWrap}`}
+                                                    >
+                                                        Model
+                                                    </TableCell>
+                                                    <TableCell
+                                                        className={`${styles.width1} ${styles.noWrap}`}
+                                                    >
+                                                        Manufacturer
+                                                    </TableCell>
+                                                    <TableCell
+                                                        className={`${styles.width1} ${styles.noWrap}`}
+                                                    >
+                                                        Qty requested
+                                                    </TableCell>
+                                                    <TableCell
+                                                        className={`${styles.width6} ${styles.noWrap}`}
+                                                    >
+                                                        Qty granted
+                                                    </TableCell>
+                                                    <TableCell
+                                                        className={`${styles.width1} ${styles.noWrap}`}
+                                                    >
+                                                        <Checkbox
+                                                            color="primary"
+                                                            data-testid={`checkall-${pendingOrder.id}`}
+                                                            onChange={(e) => {
+                                                                if (e.target.checked) {
```

Comment:
I feel like you could minimize some logic here:
```
pendingOrder.hardwareInTableRow.forEach((row) => { props.setFieldValue(<checkbox key>, e.target.checked})
```
instead of needing 2 loops and an if statement

Reply:
ohh that makes sense i think i had that if statement because i didnt have that uncheck part before